From 16489375d37dac99f51a0fdfc3101bfbbb43ccd5 Mon Sep 17 00:00:00 2001
From: Satheesh Paul <psatheesh@marvell.com>
Date: Fri, 10 Nov 2023 16:53:35 +0530
Subject: [PATCH 684/955] ci: add test to check queue index in flow regression

Add test to check queue index in flow regression.

Signed-off-by: Satheesh Paul <psatheesh@marvell.com>
Change-Id: I0e0e4384a86c7c83404bd9f3841ce44b3447ec17
Reviewed-by: Jerin Jacob <jerinj@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/115871
Reviewed-by: Kiran Kumar Kokkilagadda <kirankumark@marvell.com>
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
---
 .../flow_regression/cnxk_flow_regression.sh   | 33 +++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/marvell-ci/test/cnxk-tests/flow_regression/cnxk_flow_regression.sh b/marvell-ci/test/cnxk-tests/flow_regression/cnxk_flow_regression.sh
index f7644f1d42a19..bcb9ac03fe665 100755
--- a/marvell-ci/test/cnxk-tests/flow_regression/cnxk_flow_regression.sh
+++ b/marvell-ci/test/cnxk-tests/flow_regression/cnxk_flow_regression.sh
@@ -190,6 +190,24 @@ function testpmd_check_vlan_flags()
 	return -1
 }
 
+function testpmd_check_queue_index()
+{
+	local prefix=$1
+	local out=testpmd.out.$prefix
+	local queue_idx=$2
+
+	QUEUE_ID_COUNT=`cat $out | grep "queue=$queue_idx" | wc -l`
+
+	echo "queue occurrence count:" $QUEUE_ID_COUNT
+
+	if [[ "$QUEUE_ID_COUNT" -gt "0" ]]; then
+		return 0
+	fi
+
+	return -1
+}
+
+
 echo "Testpmd running with $TESTPMD_PORT, Coremask=$TESTPMD_COREMASK"
 testpmd_launch $PRFX \
 		" -c $TESTPMD_COREMASK -a $TESTPMD_PORT" \
@@ -315,6 +333,21 @@ fi
 
 echo "FLOW_ACTION_MARK passed"
 
+#---------------------------FLOW_ACTION_QUEUE-----------------------------------
+QUEUE_ID=0x3
+
+testpmd_test_flow $PRFX FLOW_QUEUE "flow create 0 ingress pattern ipv4 src is \
+ 10.11.12.13 / end actions  queue index $QUEUE_ID / count / end" \
+ "pcap/eth_ipv4_tcp.pcap"
+
+if ! testpmd_check_queue_index $PRFX $QUEUE_ID; then
+	echo "FAILED: incorrect queue"
+	exit 1
+fi
+
+echo "FLOW_ACTION_QUEUE passed"
+
+
 testpmd_quit  $PRFX
 echo "SUCCESS: flow regression tests completed"
 exit 0
-- 
2.25.1

